code/part-one/mining.js

```diff
@@ -23,24 +34,45 @@ class MineableTransaction {
   }
 }
 
+/**
+ * Almost identical to the non-mineable block. In fact, we'll extend it
+ * so we can reuse the calculateHash method.
+ */
 class MineableBlock extends Block {
-  /*
-    Initialize the block constructor
-
-    **Note**
-    The block should be initiated with
-      - a timestamp
-      - transactions
-      - the previous block's hash
-  */
+  /**
+   * Unlike the non-mineable block, when this one is initialized, we want the
+   * hash and nonce to not be set. This Block starts invalid, and will
+   * become valid after it is mined.
+   */
   constructor(transactions, previousHash) {
     super(transactions, previousHash);
     this.hash = '';
     this.nonce = null;
   }
 }
 
+/**
+ * The new mineable chain is a major update to our old Blockchain. We'll
+ * extend it so we can use some of its methods, but it's going to look
+ * very different when we're done.
+ */
 class MineableChain extends Blockchain {
+  /**
+   * In addition initializing a blocks array with a genesis block, this will
+   * store hard coded difficulty and reward properties. These are settings
+   * used by the mining method.
+   *
+   * Properties:
+   *   - blocks: an array of mineable blocks
+   *   - difficulty: a number, how many hex digits must be zeroed out for a
+   *     hash to be valid, this will increase mining time exponentially, so
+   *     probably best to set it pretty low (like 2 or 3)
+   *   - reward: a number, how much to award the miner of each new block
```

Comment:
should we make `difficulty` and `reward` arguments that can be passed into the `constructor` function?